### ISSUE TITLE
How to get access to DataTable instance?

### ISSUE THREAD

**Issue by greabock** [yajra/laravel-datatables#129]
Is it possible to access the DataTables instance builded with HtmlBuilder from external JS?

**Comment by yajra** [comment]
From `Datatables` you can access the `Html\Builder`, but from Builder, there is no direct access to Datatables atm. Didn't quite get what you mean by external JS.

See this example for combined usage of Datatables and Builder class. http://datatables.yajrabox.com/html/columns

**Comment by greabock** [comment]
Oh, I'm sorry. My english skill is not pretty.
Ok, look...  your builder generating script wrapped by anonymous function called by jQuery($). Like this
```
$DataTable->scripts();
```
generate next:
```
$(function(){ 
  $("#dataTableBuilder").DataTable({
        #... 
   });
});
```
but how can i get instance of DataTable JS-object to work with DataTable API? Like this
```
 var DT = $("#dataTableBuilder").DataTable({
        #... 
   });
```
now `DT` is DataTable instance.

Offcourse, i can provide myself script, but then i need to configure fields twice - in php and in js.

**Comment by yajra** [comment]
I see.. what I use to access DataTable api using external script is by using the script below:

```js
$(function(){
  var tables = $.fn.dataTable.tables(true); // get all visible DT instances
  $(tables).DataTable().search('').draw(); // perform refresh
});
```

**Comment by greabock** [comment]
Ah, thank you! This will help me! But it may be next will be more usefull? Just as variant...
```js
(function(window){
    window.LaravelDataTable = {};
    window.LaravelDataTable['nameOfInstance']  = $("#dataTableBuilder").DataTable({
        #... 
    });
})(window);
```
now we have `LaravelDataTable` collection of datatables in global namespace
```
LaravelDataTable['nameOfInstance'].search('').draw()
```

**Comment by yajra** [comment]
Oh that suggestion is good. Will try to implement that in future release. I need to learn more on JS. ^_^

**Comment by greabock** [comment]
:+1: 

little fix
```
(function(window, $){ // import "window" and "jQuery" objects
    window.LaravelDataTables['nameOfInstance']  = $("#dataTableBuilder").DataTable({
        #... 
    });
})(window, jQuery); //  provide "window" and "jQuery" objects
```

**Comment by yajra** [comment]
@greabock, request implemented on v5.10.0. A little fix though. Instance would be the ID of the table. Thanks a lot!

```js
(function(window, $){ // import "window" and "jQuery" objects
    window.LaravelDataTables =  window.LaravelDataTables || {};
    window.LaravelDataTables['dataTableBuilder']  = $("#dataTableBuilder").DataTable({
        #... 
    });
})(window, jQuery); //  provide "window" and "jQuery" objects
```

**Comment by greabock** [comment]
Ah, its magic!
